cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -257,7 +263,7 @@ func generateRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o opti
 		},
 	}
 	for _, cluster := range c.UserSecretsTargetClusters {
-		if cluster != string(api.ClusterHive) {
+		if cluster != string(api.ClusterHive) && cluster != o.clusterName {
```

Comment:
Why?

Reply:
we also generate that exact `SecretItem` in `generatePushPullSecretFrom`. I only noticed it when running in update mode with the existing build clusters.

Reply:
Ok, this might use a comment - it's not obvious from the logic itself why the code exists